### PR TITLE
Resource mng

### PR DESCRIPTION
**Pull request by PrzemekTomczyk**

### CODE REVIEW

File: ARGO/include/Game.h [hunk 1]
```diff
@@ -62,7 +70,14 @@ class Game
 	Entity m_players[4];
 	std::vector<Entity> m_entities;
 	std::vector<Entity> m_levelTiles;
+
+	Entity m_textTest1;
```
Comment (CathalRedmond) [inline ARGO/include/Game.h:73]: do tests still need to be here?

File: ARGO/include/Graphics.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <SDL.h>
+#include <stdio.h>
+#include "Utilities.h"
+
+class Graphics
+{
+public:
+	static const int SCREEN_WIDTH = Utilities::SCREEN_WIDTH;
```
Comment (CathalRedmond) [inline ARGO/include/Graphics.h]: why do you need this, can't you just use the utilities value

File: ARGO/include/Graphics.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	static const int SCREEN_HEIGHT= Utilities::SCREEN_HEIGHT;
+
+private:
+	static Graphics* s_instance;
+	static bool s_initilised;
+
+	SDL_Window* m_window;
+	SDL_Surface* m_backBuffer;
+
+public:
```
Comment (CathalRedmond) [inline ARGO/include/Graphics.h]: multiple 'public:'

Comment (CathalRedmond) [reply ARGO/include/Graphics.h]: same for 'private:'

File: ARGO/include/ResourceMng.h [hunk 4]
```diff
@@ -0,0 +1,159 @@
+//#pragma once
```
Comment (CathalRedmond) [inline ARGO/include/ResourceMng.h]: is this file needed?, its all commented out

File: ARGO/src/AssetManager.cpp [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+SDL_Texture* AssetManager::GetTexture(std::string t_filename)
+{
+	std::string fullPath = SDL_GetBasePath();
+	fullPath.append(Utilities::IMAGES_PATH + t_filename);
+
+	if (!m_textures[fullPath])
+	{
+		m_textures.at(fullPath) = loadTexture(fullPath);
```
Comment (CathalRedmond) [inline ARGO/src/AssetManager.cpp:33]: why using .at() on this line but using [] elsewhere in the function?

Comment (CathalRedmond) [reply ARGO/src/AssetManager.cpp:33]: same for get sfx function and getmusic function

File: ARGO/src/AssetManager.cpp [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+{
+	SDL_Surface* surface = TTF_RenderText_Solid(t_font, t_text.c_str(), t_colour);
+
+	if (surface == NULL)
+	{
+		printf("Text Render Error: %s\n", TTF_GetError());
+		return NULL;
+	}
+
+	SDL_Texture* tex = SDL_CreateTextureFromSurface(&m_renderer, surface);
```
Comment (CathalRedmond) [inline ARGO/src/AssetManager.cpp:102]: tex is maybe a bit ambigious as you're working with texts and textures

Comment (CathalRedmond) [reply ARGO/src/AssetManager.cpp:102]: same in load texture function

File: ARGO/src/Game.cpp [hunk 7]
```diff
@@ -233,8 +288,16 @@ void Game::preRender()
 	glm::vec2 focusPoint = glm::vec2(0, 0);
 	for (auto& player : m_players)
 	{
-		focusPoint.x += static_cast<TransformComponent*>(player.getAllComps().at(COMPONENT_ID::TRANSFORM_ID))->getPos().x;
-		focusPoint.y += static_cast<TransformComponent*>(player.getAllComps().at(COMPONENT_ID::TRANSFORM_ID))->getPos().y;
+		TransformComponent* transformComp = static_cast<TransformComponent*>(player.getAllComps().at(COMPONENT_ID::TRANSFORM_ID));
+		if (transformComp)
+		{
+			focusPoint.x += transformComp->getPos().x;
```
Comment (CathalRedmond) [inline ARGO/src/Game.cpp]: why not just have:
focusPoint += transformComp->getPos();

File: ARGO/include/AssetManager.h [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+	static AssetManager* Instance();
+	static void Release();
+
+	SDL_Texture* GetTexture(std::string t_filename);
+	Mix_Chunk* GetSfx(std::string t_filename);
+	Mix_Music* GetMusic(std::string t_filename);
+	SDL_Texture* GetText(std::string t_text, std::string t_filename, int t_size, SDL_Color t_colour);
+	TTF_Font* GetFont(std::string t_filename, int t_fontSize);
+
+	AssetManager(SDL_Renderer& t_renderer);
```
Comment (CathalRedmond) [inline ARGO/include/AssetManager.h:19]: why are the constructor/destructor below everything else?

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

File: ARGO/include/AssetManager.h (hunk 8 above)
Comment (PrzemekTomczyk) [reply ARGO/include/AssetManager.h:19]: they're meant to be private actually, forgot to change it

File: ARGO/include/Game.h (hunk 1 above)
Comment (PrzemekTomczyk) [reply ARGO/include/Game.h:73]: I'd say they can stay for now

File: ARGO/include/Graphics.h (hunk 2 above)
Comment (PrzemekTomczyk) [reply ARGO/include/Graphics.h]: forgot to remove some files, oops

File: ARGO/src/AssetManager.cpp (hunk 5 above)
Comment (PrzemekTomczyk) [reply ARGO/src/AssetManager.cpp:33]: map.at(key) throws exception when missing a key while map[key] enters a key if it doesn't exist in the map you plebian
http://www.cplusplus.com/reference/map/map/operator[]/
@CathalRedmond

File: ARGO/src/AssetManager.cpp (hunk 6 above)
Comment (PrzemekTomczyk) [reply ARGO/src/AssetManager.cpp:102]: What? they're both textures but m_text are textures for text objects while m_textures are textures for images.
"tex" is a shortened name for SDL_Texture...

File: ARGO/src/Game.cpp (hunk 7 above)
Comment (PrzemekTomczyk) [reply ARGO/src/Game.cpp]: What?? That's what it is...
